### PR TITLE
challenge(formatter): sloppy mode tests

### PR DESCRIPTION
**Pull request by victor-teles**
## Summary

fixes 
- js/identifier/parentheses/const.js
- js/sloppy-mode/function-declaration-in-if.js
- js/with/indent.js

For some cases we need to run the test using ModuleKind::Script.
I've added a allowlist to allow tests from prettier that need to run in sloppy mode.

## Test Plan

Updated snapshots

### CODE REVIEW

File: crates/biome_js_formatter/tests/prettier_tests.rs [hunk 1]
```diff
@@ -31,12 +31,30 @@ fn test_snapshot(input: &'static str, _: &str, _: &str, _: &str) {
...
         .with_indent_style(IndentStyle::Space)
         .with_indent_width(2.into());
+
     let language = language::JsTestFormatLanguage::new(source_type);
 
     let snapshot = PrettierSnapshot::new(test_file, language, options);
 
     snapshot.test()
 }
+
```
Comment (Conaclos) [inline crates/biome_js_formatter/tests/prettier_tests.rs:48]: We could add a link to the [MDN docs](https://developer.mozilla.org/en-US/docs/Glossary/Sloppy_mode) about the sloppy mode. I didn't know that sloppy mode is script/non-strict mode.

File: crates/biome_js_formatter/tests/prettier_tests.rs [hunk 2]
```diff
@@ -31,12 +31,30 @@ fn test_snapshot(input: &'static str, _: &str, _: &str, _: &str) {
...
+fn is_sloppy_mode_case(file_name: &str) -> bool {
+    let test_cases_paths = [
+        "js/with",
+        "js/sloppy-mode",
+        "js/identifier/parentheses/const.js",
+    ];
+
```
Comment (Conaclos) [inline crates/biome_js_formatter/tests/prettier_tests.rs:55]:
```suggestion
```

**Review by Conaclos — APPROVED** [review]
Looks good to me ad discussed on Discord. I left a suggestion :)

File: crates/biome_js_formatter/tests/prettier_tests.rs (hunk 1 above)
Comment (victor-teles) [reply crates/biome_js_formatter/tests/prettier_tests.rs:48]: @Conaclos Maybe is better to change it to is_non_strict_mode?

Comment (Conaclos) [reply crates/biome_js_formatter/tests/prettier_tests.rs:48]: I let you choose. Otherwise, feel free to merge :)

**Pull request merged**